webservices/ruby/style_guide.md

```diff
@@ -0,0 +1,492 @@
+      # validations
+      validates :name, presence: true
+      validates :color, presence: true
+      validates :breed, presence: true
+    end
+
+    # good
+    class Dog
+      belongs_to :user
+      belongs_to :home
+      has_many :favorite_foods
+
+      validates :name, presence: true
+      validates :color, presence: true
+      validates :breed, presence: true
+    end
+  ```
+
+### Use CamelCase for classes and modules, snake_case for variables, methods, and file names, SCREAMING_SNAKE_CASE for constants.
+  ```ruby
+    # bad
+    # app/models/LoudCamelSnakes.rb
+    class loudCamelSnakes
+      maximum_camelsnake_capacity = 450
+
+      def SneakyCamelSnake
+      end
+    end
+
+    # good
+    # app/models/loud_camel_snakes.rb
+    class LoudCamelSnakes
+      MAXIMUM_CAMELSNAKE_CAPACITY = 450
+
+      def sneaky_camel_snake
+      end
+    end
+  ```
+
+# Conditionals
+
+### Avoid lines that end with conditionals.
+  ```ruby
+    # bad
+    def sum(operand_1, operand_2)
+      operand_1 + operand_2 if valid_operands?
+    end
+
+    # good
+    def sum(operand_1, operand_2)
+      if valid_operands?
+        operand_1 + operand_2
+      end
+    end
+  ```
+
+### Avoid ternary operators.
+  ```ruby
+    # bad
+    def sum(operand_1, operand_2)
+      valid_operands? ? operand_1 + operand_2 : raise InvalidOperandError
+    end
+
+    # good
+    def sum(operand_1, operand_2)
+      if valid_operands?
+        operand_1 + operand_2
+      else
+        raise InvalidOperandError
+      end
+    end
+  ```
+
+### Prefer if over unless.
+  - use `unless` for a single negation; use `if` otherwise.
+  ```ruby
+  # bad
+    def best_pizza?
+      if !small_pizza?
+        true
+      end
+    end
+
+  # also bad
+    def best_pizza?
+      unless !tasty_pizza || small_pizza?
+        true
+      end
+    end
+
+  # good
+    def best_pizza?
+      unless small_pizza?
+        true
+      end
+    end
+
+  # also good
+    def best_pizza?
+      if tasty_pizza && !small_pizza?
+        true
+      end
+    end
+  ```
+
+# Blocks
+### Prefer &:method_name for simple method calls.
+  ```ruby
+    # bad
+    def delicious_pizza?
+      toppings.detect{ |topping| topping.delicious }
+    end
+
+    # good
+    def delicious_pizza?
+      toppings.detect(&:delicious)
+    end
+  ```
+
+### Use {...} for single-line blocks. Use do..end for multi-line blocks.
+  ```ruby
+    # bad
+      def choose_best_vacations(places)
+        places.select { |place|
+          if place.super_fun?
+            place
+          end
+        }
+      end
+
+    # also bad
+    def choose_best_vacations(places)
+      places.select do |place| place if place.super_fun? end
+    end
+
+    # good
+    def choose_best_vacations(places)
+      places.select do |place|
+        if place.super_fun?
+          place
+        end
+      end
+    end
+
+    # also good
+    def choose_best_vacations(places)
+      places.select { |place| place if place.super_fun? }
```

Comment:
does this example contradict the "Avoid lines that end with conditionals." rule above? Also, I believe just `place.super_fun?` would return the list of places, rather than needing to conditionally return the enumerated item.